File: RAT/Hades v1.6/Source/xRAT.vb/xRAT/My/MySettings.vb
Imports Microsoft.VisualBasic.ApplicationServices
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.CodeDom.Compiler
Imports System.ComponentModel
Imports System.Configuration
Imports System.Diagnostics
Imports System.Runtime.CompilerServices

Namespace xRAT.My
    <CompilerGenerated, GeneratedCode("Microsoft.VisualStudio.Editors.SettingsDesigner.SettingsSingleFileGenerator", "10.0.0.0"), EditorBrowsable(EditorBrowsableState.Advanced)> _
    Friend NotInheritable Class MySettings
        Inherits ApplicationSettingsBase
        ' Methods
        Shared Sub New()
            Class1.QaIGh5M7cuigS
            MySettings.mySettings_0 = DirectCast(SettingsBase.Synchronized(New MySettings), MySettings)
            MySettings.object_0 = RuntimeHelpers.GetObjectValue(New Object)
        End Sub

        Public Sub New()
            Class1.QaIGh5M7cuigS
        End Sub

        <EditorBrowsable(EditorBrowsableState.Advanced), DebuggerNonUserCode> _
        Private Shared Sub smethod_0(ByVal object_1 As Object, ByVal object_2 As Object)
            If Class2.Form0_0.SaveMySettingsOnExit Then
                Class11.MySettings_0.Save
            End If
        End Sub


        ' Properties
        Public Shared ReadOnly Property [Default] As MySettings
            Get
                If Not MySettings.bool_0 Then
                    Dim expression As Object = MySettings.object_0
                    ObjectFlowControl.CheckForSyncLockOnValueType(expression)
                    SyncLock expression
                        If Not MySettings.bool_0 Then
                            AddHandler Class2.Form0_0.Shutdown, New ShutdownEventHandler(AddressOf MySettings.smethod_0)
                            MySettings.bool_0 = True
                        End If
                    End SyncLock
                End If
                Return MySettings.mySettings_0
            End Get
        End Property


        ' Fields
        Private Shared bool_0 As Boolean
        Private Shared mySettings_0 As MySettings
        Private Shared object_0 As Object
    End Class
End Namespace

File: RAT/Hades v1.6/Source/xRAT.cs/xRAT/My/MySettings.cs
namespace xRAT.My
{
    using Microsoft.VisualBasic.ApplicationServices;
    using Microsoft.VisualBasic.CompilerServices;
    using System;
    using System.CodeDom.Compiler;
    using System.ComponentModel;
    using System.Configuration;
    using System.Diagnostics;
    using System.Runtime.CompilerServices;

    [CompilerGenerated, GeneratedCode("Microsoft.VisualStudio.Editors.SettingsDesigner.SettingsSingleFileGenerator", "10.0.0.0"), EditorBrowsable(EditorBrowsableState.Advanced)]
    internal sealed class MySettings : ApplicationSettingsBase
    {
        private static bool bool_0;
        private static MySettings mySettings_0;
        private static object object_0;

        static MySettings()
        {
            Class1.QaIGh5M7cuigS();
            mySettings_0 = (MySettings) SettingsBase.Synchronized(new MySettings());
            object_0 = RuntimeHelpers.GetObjectValue(new object());
        }

        public MySettings()
        {
            Class1.QaIGh5M7cuigS();
        }

        [EditorBrowsable(EditorBrowsableState.Advanced), DebuggerNonUserCode]
        private static void smethod_0(object object_1, object object_2)
        {
            if (Class2.Form0_0.SaveMySettingsOnExit)
            {
                Class11.MySettings_0.Save();
            }
        }

        public static MySettings Default
        {
            get
            {
                if (!bool_0)
                {
                    object expression = object_0;
                    ObjectFlowControl.CheckForSyncLockOnValueType(expression);
                    lock (expression)
                    {
                        if (!bool_0)
                        {
                            Class2.Form0_0.Shutdown += new ShutdownEventHandler(MySettings.smethod_0);
                            bool_0 = true;
                        }
                    }
                }
                return mySettings_0;
            }
        }
    }
}

